src/modules/network.cpp

```diff
@@ -13,7 +13,24 @@ namespace modules {
   network_module::network_module(const bar_settings& bar, string name_)
       : timer_module<network_module>(bar, move(name_)) {
     // Load configuration values
+    // m_interface could be a regex as well
     m_interface = m_conf.get(name(), "interface", m_interface);
+    std::regex interface_regex(m_interface);
+
+    struct ifaddrs* ifaddrs;
+    getifaddrs(&ifaddrs);
+
+    // If it is a regex try matching with all interfaces provided by the system
+    // The first matched interface is chosen
+    for (struct ifaddrs* i = ifaddrs; i != nullptr; i = i->ifa_next) {
+      const std::string ifname{i->ifa_name};
+      // If found set m_interface as the current
+      if (std::regex_match(ifname, interface_regex)) {
+        m_interface = ifname;
+        break;
+      }
+    }
```

Comment:
We have a similar functionality for finding a wired or wireless interface in `net::find_interface`. There, it prefers a running interface if multiple interfaces are matched. The regex matching should do the same, otherwise it gets confusing.

I think it would also be best to move this functionality into `net.cpp` next to the existing `find_interface`. And if you want and are familiar with `std::function`, you could probably make a generic `find_interface` function that takes a `std::function<bool(const ifaddrs&)>` predicate to do the interface matching in order to not duplicate too much code.

Reply:
We should probably also only run the regex matching if the given interface string is not empty (even though it won't match anything anyway, it makes the intent clearer)

Reply:
And there should be a notice log message stating that an interface was discovered through regex matching (just like when an interface was found using `interface-type`).

Reply:
> I think it would also be best to move this functionality into `net.cpp` next to the existing `find_interface`. And if you want and are familiar with `std::function`, you could probably make a generic `find_interface` function that takes a `std::function<bool(const ifaddrs&)>` predicate to do the interface matching in order to not duplicate too much code.
Im not familiar with std::function. I have some confusion about combining the argments NetType for normal find_interface and string argument for find_interface_regex.

Reply:
Then don't worry about it and make two separate functions